src/session.rs

```diff
@@ -1,7 +1,10 @@
 use tf;
-use libc::c_int;
+use libc::{c_char, c_int};
+use std::ffi::CString;
 use std::marker;
+use std::path::Path;
 use std::ptr;
+use std::result::Result as StdResult;
```

Comment:
Well, I just needed a hint on line 56 to pluck the result out, and this approach sounded elegant to me. We could consider something else though...